lib/Sema/NameBinding.cpp

```diff
@@ -40,31 +43,250 @@ using namespace swift;
 //===----------------------------------------------------------------------===//
 
 using ImportedModule = ModuleDecl::ImportedModule;
+using ImportedModuleDesc = SourceFile::ImportedModuleDesc;
 using ImportOptions = SourceFile::ImportOptions;
+using ImportFlags = SourceFile::ImportFlags;
 
-namespace {  
-  class NameBinder {    
-  public:
+struct UnboundImport {
+  SourceLoc importLoc;
+  ImportOptions options;
+  StringRef privateImportFileName;
+  Located<ImportKind> importKind;
+  ModuleDecl::AccessPathTy modulePath;
+  ModuleDecl::AccessPathTy declPath;
+
+  NullablePtr<DeclAttributes> attrs;
+  NullablePtr<ModuleDecl> underlyingModule;
+
+  /// Create an UnboundImport from an ImportDecl.
+  explicit UnboundImport(ImportDecl *ID)
```

Comment:
There’s probably enough logic in these initializers to justify pulling them further down the file.